systemvm/patches/debian/config/opt/cloud/bin/cs/CsDhcp.py

```diff
@@ -54,7 +54,8 @@ def process(self):
         self.cloud.commit()
 
         # We restart DNSMASQ every time the configure.py is called in order to avoid lease problems.
-        CsHelper.service("dnsmasq", "restart")
+        if not self.cl.is_redundant() or self.cl.is_master():
```

Comment:
Would is make sense for the ``is_redundant()`` check to be in the ``is_master()`` method?  To my way of thinking, any non-redundant router is a de facto master.  

I understand such a change may be too far to be practical at this time, so we may want to table it.